Adds changes to get latest recommendation and recommendations based on end timestamp

bharathappali opened the pull request.

dinogun commented on src/main/java/com/autotune/analyzer/services/ListRecommendation.java:
```diff
@@ -61,71 +63,145 @@ public void init(ServletConfig config) throws ServletException {
+                            error = true;
+                            sendErrorResponse(
+                                    response,
+                                    new Exception("Recommendation doesnot exist"),
```
Please use constants for all these messages. Also it should be `does not`

bharathappali commented on src/main/java/com/autotune/analyzer/utils/AnalyzerConstants.java:
```diff
@@ -545,4 +547,19 @@ private RecommendationNotificationMsgConstant() {
+
+        }
+
+        public static final String TRUE_DEFAULT = "True";
+        public static final String FALSE_DEFAULT = "False";
+        public static final String TRUE_LOWER = TRUE_DEFAULT.toLowerCase();
+        public static final String FALSE_LOWER = FALSE_DEFAULT.toLowerCase();
+        public static final String TRUE_UPPER = TRUE_DEFAULT.toUpperCase();
+        public static final String FALSE_UPPER = FALSE_DEFAULT.toUpperCase();
+        public static final String TRUE = TRUE_LOWER;
+        public static final String FALSE = FALSE_LOWER;
```
To check if `latest` param is set to true

```
if (null != latestRecommendation
                        && !latestRecommendation.isEmpty()
                        && latestRecommendation.equalsIgnoreCase(AnalyzerConstants.BooleanString.TRUE)
                ) {
                    if (!checkForTimestamp)
                        getLatest = true;
                }
```

bharathappali replied:
Not all of them are used (Currently we only use `TRUE`), I have added them so that in future we need case specific checks we can have them handy.

dinogun commented:
1. Please fix as per above comment.
2. Fix the PR check failure
3. Also, providing a non existent monitoring_end_time should throw an error. It currently ignores it (Eg. http://ip:port/listRecommendations?monitoring_end_time=2022-12-20T23:40:13.000Z)

chandrams commented:
@dino @bharathappali - I have fixed the PR check issue in #668 and it had worked previously due to a bug in listRecommendations which is now fixed.

bharathappali commented:
> 1. Please fix as per above comment.
> 
>     2. Fix the PR check failure
> 
>     3. Also, providing a non existent monitoring_end_time should throw an error. It currently ignores it (Eg. http://ip:port/listRecommendations?monitoring_end_time=2022-12-20T23:40:13.000Z)


1. Addressed the PR comments
2. I see that the failure is because the tests are looking for a experiment name which doesn't exist and it's working as expected returning `400` I'm not sure why the scripts got failed. Need to understand more to comment on it
3. It actually returns me an error saying the timestamp not found.
**Request:**

```
http://192.168.49.2:31356/listRecommendations?experiment_name=quarkus-resteasy-kruize-min-http-response-time-db_0&monitoring_end_time=2022-12-20T23:40:13.000Z
```

**Response:**

```
{
    "message": "Recommendation for timestamp - \"2022-12-20T23:40:13.000Z\" does not exist",
    "httpcode": 400,
    "documentationLink": "",
    "status": "ERROR"
}
```

dinogun commented:
@bharathappali need to resolve the conflicts as #666 was merged

bharathappali commented:
@dinogun I have added the notification if the duration based recommendation is available. I would like to request you to review the PR.

dinogun commented:
Looking much better now. However, I see this with the latest code
```
{
    "cluster_name": "cluster-one-division-bell",
    "kubernetes_objects": [
      {
        "type": "deployment",
        "name": "tfb-qrh-deployment_0",
        "namespace": "default_0",
        "containers": [
          {
            "container_image_name": "kruize/tfb-qrh:1.13.2.F_et17",
            "container_name": "tfb-server-1",
            "recommendations": {
              "notifications": [
                {
                  "type": "info",
                  "message": "Duration Based recommendations available"
                },
                {
                  "type": "info",
                  "message": "Duration Based recommendations available"
                }
              ],
              "data": {
              ...
```

We shouldn't be repeating the message.

dinogun commented:
Also API.md changes are still pending

bharathappali commented:
> Looking much better now. However, I see this with the latest code
> 
> ```
> {
>     "cluster_name": "cluster-one-division-bell",
>     "kubernetes_objects": [
>       {
>         "type": "deployment",
>         "name": "tfb-qrh-deployment_0",
>         "namespace": "default_0",
>         "containers": [
>           {
>             "container_image_name": "kruize/tfb-qrh:1.13.2.F_et17",
>             "container_name": "tfb-server-1",
>             "recommendations": {
>               "notifications": [
>                 {
>                   "type": "info",
>                   "message": "Duration Based recommendations available"
>                 },
>                 {
>                   "type": "info",
>                   "message": "Duration Based recommendations available"
>                 }
>               ],
>               "data": {
>               ...
> ```
> 
> We shouldn't be repeating the message.

@dinogun I have tested in many scenarios but I was unable to reproduce this error. Can I please know the steps to reproduce this?

bharathappali commented:
> Also API.md changes are still pending

API changes are added to the doc. Can I please have your review now?
